src/ui/components/Rating/index.js

```diff
@@ -85,20 +94,35 @@ export class RatingBase extends React.Component<InternalProps> {
     );
   };
 
+  onHoverStar = (star: number) => {
+    if (this.props.readOnly) {
+      return;
+    }
+    this.setState({ hoveringOverStar: star });
+  };
+
+  stopHovering = () => {
+    this.setState({ hoveringOverStar: null });
+  };
+
   renderRatings() {
     const { readOnly } = this.props;
     // Accept falsey values as if they are zeroes.
     const rating = this.props.rating || 0;
 
     return [1, 2, 3, 4, 5].map((thisRating) => {
+      let isSelected = thisRating - rating <= 0.25;
+      if (this.state.hoveringOverStar !== null) {
+        isSelected = thisRating <= this.state.hoveringOverStar;
+      }
       const props = {
-        className: makeClassName('Rating-choice', {
-          'Rating-selected-star': thisRating - rating <= 0.25,
+        className: makeClassName('Rating-star', `Rating-rating-${thisRating}`, {
+          'Rating-selected-star': isSelected,
           'Rating-half-star':
             thisRating - rating > 0.25 && thisRating - rating <= 0.75,
         }),
-        id: `Rating-rating-${thisRating}`,
```

Comment:
Why don't we need an `id`? Is it because we are rendering an html element and not a react component?

Reply:
The id was just used for testing. It doesn't really make sense to create an ID like this because on a page like the review listing, there could be many versions of `Rating` with `thisRating==4`.

Reply:
Oh, wait, you mean for the JSX? Ah, yes, it does need an `id`. Good catch! The linter didn't catch this because I was setting it in a props object and using a spread for the JSX constructor.

Reply:
Ignore that last comment. Heh. I was confusing `id` with `key`. Each array item already has a `key` so this will keep DOM mutations to a minimum.

Reply:
Ha, I was confusing `id` with `key` too!